video/clip.go

```diff
@@ -62,6 +62,16 @@ func getRelevantSegment(allSegments []*m3u8.MediaSegment, playHeadTime float64,
 	return 0, fmt.Errorf("error clipping: did not find a segment that falls within %v seconds", playHeadTime)
 }
 
+func ConvertUnixMillisToSeconds(firstSegment *m3u8.MediaSegment, startTimeUnixMillis, endTimeUnixMillis int64) (float64, float64) {
+	firstSegProgramDateTimeUTC := firstSegment.ProgramDateTime
```

Comment:
Is ProgramDateTime guaranteed to be set at this point or should we error or something if not present?

Reply:
Good point - if the manifest doesn't have PROGRAM-DATE-TIME tags then it should fail out here because we won't be able to calculate offsets. Will update.

Reply:
Added check for empty timestamp.